packages/core/state/selection/logics.ts

```diff
@@ -340,6 +344,53 @@ const toggleFileFolderCollapse = createLogic({
     type: [TOGGLE_FILE_FOLDER_COLLAPSE],
 });
 
+/**
+ * Interceptor responsible for transforming COLLAPSE_ALL_FILE_FOLDERS and EXPAND_ALL_FILE_FOLDERS
+ * actions into SET_OPEN_FILE_FOLDERS actions by either setting to none or recursively
+ * unpacking the directory structure
+ */
+const toggleAllFileFolders = createLogic({
+    async process(deps: ReduxLogicDeps, dispatch, done) {
+        const { action, getState } = deps;
+        const hierarchy = selection.selectors.getAnnotationHierarchy(getState());
+        const annotationService = interaction.selectors.getAnnotationService(getState());
+        const selectedFileFilters = selection.selectors.getFileFilters(getState());
+        const fileFoldersToOpen: FileFolder[] = [];
+        dispatch(setIsLoadingFullTree(true) as AnyAction);
+        // Recursive helper
+        async function unpackAllFileFolders(values: string[], pathSoFar: string[]) {
+            for (const value of values) {
+                fileFoldersToOpen.push(new FileFolder([...pathSoFar, value] as AnnotationValue[]));
+
+                // At end of folder hierarchy
+                if (!!hierarchy.length && pathSoFar.length === hierarchy.length - 1) continue;
+
+                const childNodes = await annotationService.fetchHierarchyValuesUnderPath(
+                    hierarchy,
+                    [...pathSoFar, value],
+                    selectedFileFilters
+                );
+                if (childNodes.length) {
+                    // Not a leaf
+                    unpackAllFileFolders(childNodes, [...pathSoFar, value]);
+                }
+            }
+        }
+
+        if (action.type === EXPAND_ALL_FILE_FOLDERS) {
+            const rootHierarchyValues = await annotationService.fetchRootHierarchyValues(
+                hierarchy,
+                selectedFileFilters
+            );
+            await unpackAllFileFolders(rootHierarchyValues, []);
+        }
+        dispatch(setOpenFileFolders(fileFoldersToOpen));
+        dispatch(setIsLoadingFullTree(false) as AnyAction);
+        done();
+    },
+    type: [COLLAPSE_ALL_FILE_FOLDERS, EXPAND_ALL_FILE_FOLDERS],
```

Comment:
I think adding `COLLAPSE_ALL_FILE_FOLDERS` has complicated this logic and is unnecessary because unlike `EXPAND_ALL_FILE_FOLDERS` there is no async action that needs to transpire. Instead, for `COLLAPSE_ALL_FILE_FOLDERS` you can just have a reducer that sets `openFileFolders: []` in the state.

Reply:
resolved in [c16c1ae](https://github.com/AllenInstitute/biofile-finder/pull/446/commits/c16c1ae522161412502360a675bab15dbacbf44f)